Sage-Bionetworks/schematic
# Retain formatting of Display name for annotations and table column names

## Summary
Retain display name formatting when submitting manifests, for column labels on tables, and and adding annotations to linked files.
 
**Note: the testing procedure has been updated to conform to new parameters designed and discussed in comments below**

This is to address [FDS-1199](https://sagebionetworks.jira.com/browse/FDS-1199). 

## To test

1. In a testing project, create a new dataset folder, make note of the folder name, and get record its Synapse ID. 
2. Make sure the dataset is in a Fileview and record the Fileview Synapse ID.
3. Download the data files from this synapse test [project](https://www.synapse.org/#!Synapse:syn53019444). 
Put these files and put into your new test dataset folder.
4. Download the the Bulk RNA seq manifest from the same dataset folder, and save locally to submit via schematic
5. Open the manifest in and editor.
6. Remove the UUID/Id and entitityId columns prior to inital submission or it will be pointing to the wrong entityId columns
7. Update the Filepaths in the `Filename` column to point to the new path the files are located in on synapse
4. Submit test manifest to this dataset as appropriate to fully test (see below) via API and CLI. **IMPORTANT**In between tests make sure to delete any generated table, and annotations (on the data files)to ensure the table column names and annotation keys are updated as expected.

### Example CLI command:

To `config.yml`:
- Update filview as needed,
- Update data model path to the `example.model.jsonld`


```
schematic model -c config.yml submit -mp path/to/synapse_storage_manifest_bulkrna-seqassay.csv -d dataset_id -mrt table_and_file --table_column_names display_label --annotation_keys display_label
```

### Example API input:
```
schema_url: https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld
dataset_id: synID
manifest_record_type: table_and_file
restrict_rules: false
asset_view: synID
table_column_names: display_label
annotation_keys: display_label
```

## Check outputs

- Look at the annotations on the files in the test dataset.
  - Should see `sampleID` recorded as an annotation. If compared to the manifest, you will see that the `-`'s have been strippped along with the lowercase first letter.
  - The manifest file should have a column named `sample-I-D`
- Look at the table `bulkrna-seqassay_synapse_storage_manifest_table` in your project. Should see it has a column named `sampleID`.

These changes are significant bc normally, names given would be `SampleID`.

Delete annotations and table, then submit again with other options as described in the comment below this one.


[FDS-1199]: https://sagebionetworks.jira.com/browse/FDS-1199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
